fishtown-analytics/docs.getdbt.com
# Bug/versioned toc

## What are you changing in this pull request and why?


Re:[this Asana task](https://app.asana.com/0/1200099998847559/1204106336641335/f), this PR resolves a glitch in the way the Table of Contents data was generated, which caused the TOC to be incorrect when a user would refresh on a page with versioned sections.

[Explanation of bug here.](https://github.com/dbt-labs/docs.getdbt.com/pull/3419/files#r1204719433)

## To test
- Open a docs page which includes versioned content, such as [this page](https://deploy-preview-3419--docs-getdbt-com.netlify.app/reference/node-selection/methods), focusing on the The "file" or "fqn" method section.
- Switch the version in the nav to v1.1 or lower.
- Refresh the page
    - [**On production,**](https://docs.getdbt.com/reference/node-selection/methods) The `The "file" or "fqn" method` item remains in the ToC on the right, even when the section is not on the page. This section is only available on v1.2 or greater.
    - [**On the preview of this PR,**](https://deploy-preview-3419--docs-getdbt-com.netlify.app/reference/node-selection/methods) the `The "file" or "fqn" method` section does not appear in the TOC on page refresh.


## Checklist

- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [x] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."
